File: 7kyu/(7 kyu) Shortest Word/(7 kyu) Shortest Word.go
package kata

import "strings"

func FindShort(s string) int {
  maxLen := len(s)
  words := strings.Fields(s)
  for _, word := range(words) {
    if len(word) <= maxLen {
      maxLen = len(word)
    }
  }
  
  return maxLen
}

File: 7kyu/(7 kyu) Shortest Word/(7 kyu) Shortest Word.cs
// #1 Plain Solution
public class Kata
{
  public static int FindShort(string s)
  {
    string[] words = s.Split(' ');
    int min = s.Length;
    foreach (string word in words)
    {
        min = word.Length < min ? word.Length : min;
    }
    return min;
  }
}

// #2 Optimized solution
using System.Linq;

public class Kata
{
  public static int FindShort(string s)
  {
    return s.Split(' ').Min(x => x.Length);
  }
}
